--- DirectFB-0.9.21/gfxdrivers/cle266/Makefile.am.orig	2004-10-09 01:44:31.000000000 +0200
+++ DirectFB-0.9.21/gfxdrivers/cle266/Makefile.am	2005-01-15 15:07:27.813759856 +0100
@@ -32,7 +32,7 @@
 libdirectfb_cle266_la_LDFLAGS = \
 	-export-dynamic		\
 	-avoid-version		\
-	$(DFB_LDFLAGS)
+	$(DFB_LDFLAGS) -lm
 
 
 include $(top_srcdir)/rules/libobject.make
--- DirectFB-0.9.21/gfxdrivers/unichrome/Makefile.am.orig	2004-10-09 01:48:19.000000000 +0200
+++ DirectFB-0.9.21/gfxdrivers/unichrome/Makefile.am	2005-01-15 15:07:38.104195472 +0100
@@ -34,7 +34,7 @@
 libdirectfb_unichrome_la_LDFLAGS = \
 	-export-dynamic		   \
 	-avoid-version		   \
-	$(DFB_LDFLAGS)
+	$(DFB_LDFLAGS) -lm
 
 
 include $(top_srcdir)/rules/libobject.make
--- DirectFB-1.0.0.orig/gfxdrivers/matrox/matrox_maven.c	2007-01-27 23:22:14.000000000 +0100
+++ DirectFB-1.0.0/gfxdrivers/matrox/matrox_maven.c	2007-03-07 12:45:56.000000000 +0100
@@ -37,7 +37,6 @@
 #include <sysfs/libsysfs.h>
 #endif
 
-#include <linux/i2c.h>
 #include <linux/i2c-dev.h>
 
 #include <directfb.h>
--- DirectFB-1.0.0.orig/include/dfb_types.h	2007-03-03 21:36:07.000000000 +0100
+++ DirectFB-1.0.0/include/dfb_types.h	2007-03-09 11:43:58.000000000 +0100
@@ -32,6 +32,7 @@
 
 #ifdef USE_KOS
 
+#include <asm/types.h>
 #include <sys/types.h>
 
 typedef uint8 u8;
