--- DirectFB-1.4.15/lib/direct/ppcasm_memcpy_cachable.S.orig	2011-09-29 11:51:21.000000000 +0200
+++ DirectFB-1.4.15/lib/direct/ppcasm_memcpy_cachable.S	2011-12-30 19:46:23.464488297 +0100
@@ -34,16 +34,8 @@
 
 #define __ASSEMBLY__
 
-#if defined(CONFIG_8xx) || defined(CONFIG_403GCX)
-#define L1_CACHE_LINE_SIZE       16
-#define LG_L1_CACHE_LINE_SIZE     4 
-#elif defined(CONFIG_PPC64BRIDGE) 
-#define L1_CACHE_LINE_SIZE      128
-#define LG_L1_CACHE_LINE_SIZE     7  
-#else
 #define L1_CACHE_LINE_SIZE       32
 #define LG_L1_CACHE_LINE_SIZE     5
-#endif
 
 #include "ppc_asm.h"
 
@@ -140,9 +132,7 @@
 	mtctr	r0
 	beq	63f
 53:
-#if !defined(CONFIG_8xx)
 	dcbz	r11,r6
-#endif
 	COPY_16_BYTES
 #if L1_CACHE_LINE_SIZE >= 32
 	COPY_16_BYTES
