--- DirectFB-0.9.14/configure.in.orig	Mon Oct 28 19:20:06 2002
+++ DirectFB-0.9.14/configure.in	Thu Nov  7 11:21:17 2002
@@ -64,6 +64,7 @@
 AC_HEADER_STDC
 AC_C_CONST
 AC_C_BIGENDIAN
+AM_PROG_AS
 
 AC_PATH_PROGS(PERL, perl5 perl)
 
@@ -230,9 +231,8 @@
 if test "x$enable_debug" = xyes; then
     AC_DEFINE(DFB_DEBUG,1,[Define to 1 if you want debugging support.])
     DFB_INTERNAL_CFLAGS="$DFB_INTERNAL_CFLAGS -DDFB_DEBUG"
-    DFB_CFLAGS="$DFB_CFLAGS -O3 -g3 -fno-inline"
+    DFB_CFLAGS="$DFB_CFLAGS -fno-inline"
 else
-    DFB_CFLAGS="$DFB_CFLAGS -O3 -g"
     if test "x$enable_profiling" = xno; then
         DFB_CFLAGS_OMIT_FRAME_POINTER="-fomit-frame-pointer"
     fi
