--- DirectFB-0.9.18/configure.in.orig	2003-04-24 14:12:15.000000000 +0200
+++ DirectFB-0.9.18/configure.in	2003-07-12 14:34:09.000000000 +0200
@@ -67,6 +67,7 @@
 AC_HEADER_STDC
 AC_C_CONST
 AC_C_BIGENDIAN
+AM_PROG_AS
 
 AC_PATH_PROGS(PERL, perl5 perl)
 
@@ -88,13 +89,6 @@
 AM_CONDITIONAL(FBDEV_CORE, test x$enable_fbdev = xyes)
 
 
-dnl Clear default CFLAGS
-if test x"$CFLAGS" = x"-g -O2"; then
-  CFLAGS=
-fi
-
-CFLAGS="-O3 -pipe $CFLAGS"
-
 if test "x$GCC" = xyes; then
   CFLAGS="-Wall $CFLAGS"
 fi
@@ -258,6 +252,7 @@
 if test "x$enable_debug" = xyes; then
     DFB_INTERNAL_CFLAGS="$DFB_INTERNAL_CFLAGS -DDFB_DEBUG"
     CFLAGS="$CFLAGS -g3 -fno-inline"
+    DFB_CFLAGS_OMIT_FRAME_POINTER=""
 fi
 AM_CONDITIONAL(ENABLE_DEBUG, test "x$enable_debug" = xyes)
 
