--- DirectFB-0.9.25.1/wm/unique/Makefile.am~	2006-05-03 10:22:38.000000000 +0300
+++ DirectFB-0.9.25.1/wm/unique/Makefile.am	2006-11-07 09:27:04.794005664 +0200
@@ -19,7 +19,7 @@
 uwmdump_SOURCES = uwmdump.c
 
 uwmdump_LDADD = \
-	libuniquewm.la					\
+	$(top_builddir)/wm/unique/libuniquewm.la	\
 	$(top_builddir)/lib/direct/libdirect.la		\
 	$(top_builddir)/lib/fusion/libfusion.la		\
 	$(top_builddir)/src/libdirectfb.la
@@ -28,7 +28,7 @@
 test_color_SOURCES = test_color.c
 
 test_color_LDADD = \
-	libuniquewm.la					\
+	$(top_builddir)/wm/unique/libuniquewm.la	\
 	$(top_builddir)/lib/direct/libdirect.la		\
 	$(top_builddir)/lib/fusion/libfusion.la		\
 	$(top_builddir)/src/libdirectfb.la
@@ -37,7 +37,7 @@
 test_foo_SOURCES = test_foo.c
 
 test_foo_LDADD = \
-	libuniquewm.la					\
+	$(top_builddir)/wm/unique/libuniquewm.la	\
 	$(top_builddir)/lib/direct/libdirect.la		\
 	$(top_builddir)/lib/fusion/libfusion.la		\
 	$(top_builddir)/src/libdirectfb.la
@@ -46,7 +46,7 @@
 stret_test_SOURCES = stret_test.c
 
 stret_test_LDADD = \
-	libuniquewm.la					\
+	$(top_builddir)/wm/unique/libuniquewm.la	\
 	$(top_builddir)/lib/fusion/libfusion.la		\
 	$(top_builddir)/lib/direct/libdirect.la		\
 	$(top_builddir)/src/libdirectfb.la
@@ -69,7 +69,7 @@
 libdirectfbwm_unique_la_LIBADD = \
 	$(top_builddir)/lib/direct/libdirect.la \
 	$(top_builddir)/lib/fusion/libfusion.la \
-	libuniquewm.la
+	$(top_builddir)/wm/unique/libuniquewm.la
 
 libdirectfbwm_unique_la_LDFLAGS = \
 	-avoid-version	\
