--- DirectFB-0.9.24/include/dfb_types.h.orig	2005-07-26 22:58:50.000000000 +0200
+++ DirectFB-0.9.24/include/dfb_types.h	2006-01-26 21:09:57.021772000 +0100
@@ -5,15 +5,7 @@
 
 #ifndef USE_KOS
 
-#define __u8  u_int8_t
-#define __u16 u_int16_t
-#define __u32 u_int32_t
-#define __u64 u_int64_t
-
-#define __s8  int8_t
-#define __s16 int16_t
-#define __s32 int32_t
-#define __s64 int64_t
+#include <asm/types.h>
 
 #else
 
--- DirectFB-0.9.24/include/directfb.h.orig	2005-10-24 18:29:11.000000000 +0200
+++ DirectFB-0.9.24/include/directfb.h	2006-01-27 21:55:51.645587750 +0100
@@ -846,7 +846,7 @@
                                            any character advance of fixed or
                                            proportional fonts */
      DFDESC_FRACT_HEIGHT = 0x00000020,  /* fractional height is set */
-     DFDESC_FRACT_WIDTH  = 0x00000040,  /* fractional width is set */
+     DFDESC_FRACT_WIDTH  = 0x00000040   /* fractional width is set */
 } DFBFontDescriptionFlags;
 
 /*
