--- DirectFB-0.9.19/gfxdrivers/i810/i810.c.orig	2003-08-01 20:00:44.000000000 +0200
+++ DirectFB-0.9.19/gfxdrivers/i810/i810.c	2003-08-01 19:59:36.000000000 +0200
@@ -26,6 +26,7 @@
 #include <stdio.h>
 #include <unistd.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include <sys/mman.h>
 #include <fcntl.h>
@@ -285,7 +286,7 @@
 						   state->color.b);
 		break;
 	case DSPF_RGB24:
-		i810drv->color_value = PIXEL_RGB24(state->color.r,
+		i810drv->color_value = PIXEL_RGB32(state->color.r,
 						   state->color.g,
 						   state->color.b);
 		break;
--- DirectFB-0.9.19/gfxdrivers/i810/i810_overlay.c.orig	2003-08-01 20:05:32.000000000 +0200
+++ DirectFB-0.9.19/gfxdrivers/i810/i810_overlay.c	2003-08-01 20:05:38.000000000 +0200
@@ -391,7 +391,7 @@
 		i810ovl->regs->dclrkv = PIXEL_RGB16(r, g, b);
 		break;
 	case DSPF_RGB24:
-		i810ovl->regs->dclrkv = PIXEL_RGB24(r, g, b);
+		i810ovl->regs->dclrkv = PIXEL_RGB32(r, g, b);
 		break;
 	default:
 		BUG("unexpected pixelformat");
