--- DirectFB-1.2.3/include/directfb.h.orig	2008-08-19 11:25:16.000000000 +0200
+++ DirectFB-1.2.3/include/directfb.h	2008-09-03 17:34:21.868983118 +0200
@@ -970,7 +970,7 @@
                                            any character advance of fixed or
                                            proportional fonts */
      DFDESC_FRACT_HEIGHT = 0x00000020,  /* fractional height is set */
-     DFDESC_FRACT_WIDTH  = 0x00000040,  /* fractional width is set */
+     DFDESC_FRACT_WIDTH  = 0x00000040   /* fractional width is set */
 } DFBFontDescriptionFlags;
 
 /*
@@ -3108,7 +3108,7 @@
      DSPD_SRC_ATOP       =  9, /* fs: da      fd: 1.0-sa */
      DSPD_DST_ATOP       = 10, /* fs: 1.0-da  fd: sa     */
      DSPD_ADD            = 11, /* fs: 1.0     fd: 1.0    */
-     DSPD_XOR            = 12, /* fs: 1.0-da  fd: 1.0-sa */
+     DSPD_XOR            = 12  /* fs: 1.0-da  fd: 1.0-sa */
 } DFBSurfacePorterDuffRule;
 
 /*
